lib/std/fs.zig

```diff
@@ -1469,9 +1469,11 @@ pub const Dir = struct {
         while (true) {
             self.makeDir(component.path) catch |err| switch (err) {
                 error.PathAlreadyExists => {
-                    // TODO stat the file and return an error if it's not a directory
+                    // stat the file and return an error if it's not a directory
                     // this is important because otherwise a dangling symlink
                     // could cause an infinite loop
+                    var fstat = try self.statFile(sub_path);
+                    if(fstat.kind != .directory){ return error.PathIsNotDir; }
```

Comment:
```suggestion
                    if (fstat.kind != .directory) return error.NotDir;
```

to match the named used in other fs error sets.